pyop2/mpi.py

```diff
@@ -272,10 +281,12 @@ def __exit__(self, exc_type, exc_value, traceback):
         pass
 
 
-def internal_comm(comm):
+def internal_comm(comm, obj):
     """ Creates an internal comm from the user comm.
     If comm is None, create an internal communicator from COMM_WORLD
     :arg comm: A communicator or None
+    :arg obj: The object which the comm is an attribute of
```

Comment:
I don't love this pattern. Would something like a mixin class be better?

Reply:
Not my favourite either. I agree this should derive from an `MPIObject` base class, but the place for that to live is either `mpi4py` or a dedicated add on library and this is a future task. I'd rather not refactor all the PyOP2 objects (again).